### PR TITLE
Filter private methods from tab completion

### PR DESCRIPTION
**Pull request by katlings**
Issue #1636 

I built and installed this on my CPX and it works the way I expected:

```
Adafruit CircuitPython 4.0.0-rc.1-23-g54aa1ce6d on 2019-05-06; Adafruit CircuitPlayground Express with samd21g18
>>> class Test:
...     def _internal(self):
...         return 1
...     
...     def external(self):
...         return 2
...     
...     def external_foo(self):
...         return 3
...     
...     def foo(self):
...         return 4
...         
...         
... 
>>> Test.
external        external_foo    foo
>>> Test.external
external        external_foo
>>> Test._
__class__       __module__      __name__        __qualname__
_internal
>>> Test.__
__class__       __module__      __name__        __qualname__
```

### CODE REVIEW

**Comment by katlings** [comment]
FWIW, this also allows the special case for autocompleting "`import `" when pressing tab at the start of a line to work (previously for me, it started with `__name__`)!

File: py/repl.c [hunk 1]
```diff
@@ -187,6 +187,11 @@ size_t mp_repl_autocomplete(const char *str, size_t len, const mp_print_t *print
                 if (s_len <= d_len && strncmp(s_start, d_str, s_len) == 0) {
                     mp_load_method_protected(obj, q, dest, true);
                     if (dest[0] != MP_OBJ_NULL) {
+                        // special case; filter out words that begin with underscore
+                        // unless there's already a partial match
+                        if (s_len == 0 && *d_str == '_') {
```
Comment (deshipu) [inline py/repl.c]: nitpick: would ``d_str[0]`` be more readable here?

Comment (katlings) [reply py/repl.c]: I haven't programmed C in years lol, forgot that was an option. I like making the first character check explicit; will update soon

**Comment by katlings** [comment]
This is ready! :D

**Pull request merged**